### PR TITLE
Add basic eslint prettier stuff

### LINKED ISSUE #1: Missing basic setup with guard rails
At minimum, it should have TS ESLint, Prettier etc. and pre-commit hook to validate & correct some of these violations

### PR DESCRIPTION
**Pull request by crimelabs786**
Closes #1

### CODE REVIEW

**Comment by crimelabs786** [comment]
Also relevant to this: https://github.com/indiainvestments/indiainvestments.github.io/projects/5#card-54416251

File: package.json [hunk 1]
```diff
@@ -8,13 +8,24 @@
...
+    "eslint": "^7.19.0",
+    "eslint-config-prettier": "^7.2.0",
+    "eslint-plugin-prettier": "^3.3.1",
+    "faker": "^5.3.1",
     "husky": "^4.3.8",
+    "jest": "^26.6.3",
+    "prettier": "^2.2.1",
+    "pretty-quick": "^3.1.0",
+    "ts-jest": "^26.5.0",
+    "ts-node": "^9.1.1",
```
Comment (zulienthegreat) [inline package.json:23]: No need for this. `ncc run` can run typescript files.

Comment (crimelabs786) [reply package.json:23]: Ahh ok, will remove

**Comment by crimelabs786** [comment]
@zulienthegreat because we don't have a CI action for PR validation here, you should check out my branch locally, test it out manually, and review on the basis of that

**Comment by zulienthegreat** [comment]
Add the previous tsconfig with new changes on top of them. Remove the changes you made to existing configuration settings. No need for that.

Also, could you please add the lint-stage command instead of a script. You haven't given any reason for that approach. That approach may have problem with cross compaitibility in future (windows....)

**Review by zulienthegreat — APPROVED** [review]

**Pull request merged**